AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/AWSPinpointAnalyticsPlugin+ClientBehavior.swift

```diff
@@ -22,16 +22,13 @@ extension AWSPinpointAnalyticsPlugin {
             currentEndpointProfile.addUserProfile(userProfile)
         }
 
-        pinpoint.update(currentEndpointProfile).continueWith { (task) -> Any? in
-            guard task.error == nil else {
-                // TODO: some error mapping
-                let error = task.error! as NSError
-                Amplify.Hub.dispatchIdentifyUser(AnalyticsErrorHelper.getDefaultError(error))
-                return nil
+        Task {
+            do {
+                try await pinpoint.update(currentEndpointProfile)
+                Amplify.Hub.dispatchIdentifyUser(identityId, userProfile: userProfile)
+            } catch {
+                Amplify.Hub.dispatchIdentifyUser(AnalyticsErrorHelper.getDefaultError(error as NSError))
```

Comment:
nit: can we have an override of `AnalyticsErrorHelper.getDefaultError` that accepts a native Swift error instead of casting to `NSError` in multiple places? it can be probably used by other categories

Reply:
Sure, I'll add that overload to `AnalyticsErrorHelper`